devito/types/basic.py

```diff
@@ -1450,12 +1454,19 @@ def indexify(self, indices=None, subs=None):
         # Indices after substitutions
         indices = []
         for a, d, o, s in zip(self.args, self.dimensions, self.origin, subs):
-            if d in a.free_symbols:
+            if a.is_Function and len(a.args) == 1:
+                # E.g. Abs(expr)
```

Comment:
it has to be an integer function, so maybe you can check for Abs; if not Abs but a Function, return an error?

I don't know, maybe nitpicking, feel free to ignore...

Reply:
Could be floor/ceil/INT/...., so lots of options, this just check it's unary Op, multi-arg op are too weird here so maybe should raise an error